Sources/EduIDExpansion/Controllers/Onboarding/EnterPersonalInfoViewController.swift

```diff
@@ -1,24 +1,59 @@
 import UIKit
 import TinyConstraints
 
-class EnterPersonalInfoViewController: EduIDBaseViewController, ValidatedTextFieldDelegate {
+class EnterPersonalInfoViewController: EduIDBaseViewController {
+    
+    private var viewModel: EnterPersonalInfoViewModel
+    private var keyboardHeight: CGFloat?
+    private var isKeyBoardOnScreen = false
+    private let inset: CGFloat = 24
     
     var stack: AnimatedVStackView!
     let requestButton = EduIDButton(type: .primary, buttonTitle: "Request you eduID")
+    let emailField = TextStackViewWithValidation(title: "Your email address", placeholder: "e.g. timbernerslee@gmail.com", keyboardType: .emailAddress)
+    let scrollView = UIScrollView()
     
-    var validationMap: [Int: Bool] = [0:false, 1:false, 2: false] {
-        didSet {
-            var isTrue = true
-            validationMap.forEach({ (key: Int, value: Bool) in
-                if !value {
-                    isTrue = false
-                }
-            })
-            requestButton.isEnabled = isTrue
+    //MARK: - spacing
+    let spacingView = UIView()
+    
+    
+    //MARK: - init
+    init(viewModel: EnterPersonalInfoViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+        
+        viewModel.makeNextTextFieldFirstResponderClosure = { [weak self] tag in
+            guard tag != 3 else {
```

Comment:
Avoid using magic values. Prefer defining a constant that describes with filed this is. For example `static let emailFieldTag = 3`.

Reply:
I did this